src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -607,50 +741,44 @@ public static async ValueTask<FormulaValue> AverageTable(EvalVisitor runner, Eva
                 return CommonErrors.DivByZeroError(irContext);
             }
 
-            return await RunAggregatorAsync("Average", new AverageAgg(), runner, context, irContext, args).ConfigureAwait(false);
+            return await RunAggregatorAsync("Average", irContext.ResultType == FormulaType.Decimal ? new AverageDecimalAgg() : new AverageAgg(), runner, context, irContext, args);
         }
 
         // https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-mod
         public static FormulaValue Mod(IRContext irContext, FormulaValue[] args)
         {
-            double arg0 = ((NumberValue)args[0]).Value;
-
-            if (args[1] is TableValue table)
+            if (irContext.ResultType == FormulaType.Number)
             {
-                List<FormulaValue> modResults = new List<FormulaValue>();
-
-                foreach (DValue<RecordValue> row in table.Rows)
-                {
-                    FormulaValue val = row.Value.GetField("Value");
-                    double arg1 = val is NumberValue nv ? nv.Value : 0d;
-                    modResults.Add(ModInternal(arg0, arg1, IRContext.NotInSource(FormulaType.Number)));
-                }
-
-                return new InMemoryTableValue(irContext, StandardTableNodeRecords(irContext, modResults.ToArray(), forceSingleColumn: true));
+                return ModFloat(irContext, (NumberValue)args[0], (NumberValue)args[1]);
+            }
+            else if (irContext.ResultType == FormulaType.Decimal)
+            {
+                return ModDecimal(irContext, (DecimalValue)args[0], (DecimalValue)args[1]);
             }
             else
             {
-                double arg1 = ((NumberValue)args[1]).Value;
-
-                return ModInternal(arg0, arg1, irContext);
+                return CommonErrors.UnreachableCodeError(irContext);
             }
         }
 
-        private static FormulaValue ModInternal(double arg0, double arg1, IRContext irContext)
+        public static FormulaValue ModFloat(IRContext irContext, NumberValue arg0nv, NumberValue arg1nv)
         {
+            double arg0 = arg0nv.Value;
+            double arg1 = arg1nv.Value;
+
             if (arg1 == 0)

```

Comment:
>if (arg1 == 0) [](http://example.com/codeflow?start=12&length=14)

even if absolutely not important, i'd add a comment that double.NegativeZero is also considered here

Reply:
Done.